bindings/python/pymongocrypt/synchronous/credentials.py

```diff
@@ -72,11 +75,14 @@ def _get_azure_credentials():
     url += "?api-version=2018-02-01"
     url += "&resource=https://vault.azure.net"
     headers = { "Metadata": "true", "Accept": "application/json" }
+    client = httpx.Client()
     try:
-        response = requests.get(url, headers=headers)
+        response = client.get(url, headers=headers)
     except Exception as e:
         msg = "Failed to acquire IMDS access token: %s" % e
         raise MongoCryptError(msg)
+    finally:
+        client.close()
```

Comment:
We used to implicitly use request's process wide connection pooling but now we're opening a new connection for each request. Is this going to have a negative perf impact?

Reply:
This might not matter because libmongocrypt caches the creds so this method is called infrequently.